packages/cli/src/cms/api/middlewares/sunset.middleware.ts

```diff
@@ -0,0 +1,68 @@
+import type { Response } from 'node-fetch';
+
+export type SunsetWarning = { sunsetDate: Date; isSunsetExpired: boolean; warning: string };
+export type SunsetWarningsBuffer = SunsetWarning[];
+
+type SunsetMiddlewareOptions =
+  | {
+      bufferSunsets: true;
+    }
+  | {
+      onSunset: (arg: SunsetWarning) => void;
+    };
+
+export function createApiSunsetMiddleware(options: SunsetMiddlewareOptions) {
+  const buffer: SunsetWarningsBuffer = [];
+
+  function getSunsetDate(response: Response): number | undefined {
+    let sunsetDate = response.headers.get('sunset');
+
+    if (sunsetDate) {
+      return Date.parse(sunsetDate);
+    }
+
+    sunsetDate = response.headers.get('Sunset');
+
+    if (sunsetDate) {
+      return Date.parse(sunsetDate);
+    }
+
+    return;
+  }
+
+  return {
+    middleware: (response: Response) => {
+      const sunsetTime = getSunsetDate(response);
+
+      if (sunsetTime) {
+        const sunsetDate = new Date(sunsetTime);
+        let sunsetWarning: SunsetWarning;
+
+        if (sunsetTime > Date.now()) {
+          sunsetWarning = {
+            sunsetDate,
+            isSunsetExpired: false,
+            warning: `Endpoint ${
```

Comment:
Where does this content go to? We usually don't use "endpoint" in customer facing language, or expose the underlying URL. If you can put together some examples of the expected output, I'm happy to help with re-wording.

Reply:
This content will be shown to user at the end of `push` or `push-status` command logs. I've updated the description and added screenshots to explain intentions of this functionality.

> We usually don't use "endpoint" in customer facing language, or expose the underlying URL.

Would be better to use "resource" in this case?

Reply:
I'm not sure! What are we actually trying to say?

Reply:
We're going to notify user that some endpoint used by `push` or `push-status` command will not be/is not compatible with current CLI version due to breaking change or endpoint shutdown.